src/components/WeatherTrends.tsx

```
@@ -0,0 +1,151 @@
+import { LineChart, Line, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts';
+import { History, Calendar } from 'lucide-react';
+import { format, fromUnixTime } from 'date-fns';
+import { ForecastData } from '../types';
+import { kelvinToCelsius } from '../utils';
+
+interface WeatherTrendsProps {
+  forecast: ForecastData;
+  isDark: boolean;
+}
+
+const generateChartData = (forecastList: any[], timeframe: 'daily' | 'weekly') => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)